test/integration/instance/destroy.test.js

```diff
@@ -76,24 +102,254 @@ describe(Support.getTestDialectTeaser('Instance'), () => {
+        destroyTime: { type: Support.Sequelize.DATE, field: 'destroy_time' }
+      }, { paranoid: true, deletedAt: 'destroyTime' });
+
+      return ParanoidUser.sync({ force: true }).then(() => {
+        return ParanoidUser.create({
+          username: 'username'
+        });
+      }).then(user => {
+        return user.destroy();
+      }).then(user => {
+        expect(user.dataValues.destroyTime).to.be.ok;
+        expect(user.dataValues.destroy_time).to.not.be.ok;
+
+        return ParanoidUser.findOne({
+          paranoid: false,
+          where: {
+            username: 'username'
+          }
+        });
+      }).then(user => {
+        expect(user).to.be.ok;
+        expect(user.destroyTime).to.be.ok;
+        expect(user.destroy_time).to.not.be.ok;
+      });
+    });
+
+    it('persists other model changes when soft deleting', function() {
+      const ParanoidUser = this.sequelize.define('ParanoidUser', {
+        username: Support.Sequelize.STRING
+      }, { paranoid: true });
+
+      let deletedAt;
+      return ParanoidUser.sync({ force: true }).then(() => {
+        return ParanoidUser.create({
+          username: 'username'
+        });
+      }).then(user => {
+        user.username = 'foo';
+        return user.destroy();
+      }).then(user => {
+        expect(user.username).to.equal('foo');
+        expect(user.deletedAt).to.be.ok;
+        deletedAt = user.deletedAt;
+
+        return ParanoidUser.findOne({
+          paranoid: false,
+          where: {
+            username: 'foo'
+          }
+        });
+      }).tap(user => {
+        expect(user).to.be.ok;
+        expect(user.deletedAt.toISOString()).to.equal(deletedAt.toISOString());
+        expect(user.username).to.equal('foo');
+      }).then(user => {
+        // update model and delete again
+        user.username = 'bar';
+        return user.destroy();
+      }).then(user => {
+        expect(user.deletedAt.toISOString()).to.equal(deletedAt.toISOString(),
+          'should not updated deletedAt when destroying multiple times');
+
+        return ParanoidUser.findOne({
+          paranoid: false,
+          where: {
+            username: 'bar'
+          }
+        });
+      }).then(user => {
+        expect(user).to.be.ok;
+        expect(user.deletedAt.toISOString()).to.equal(deletedAt.toISOString());
+        expect(user.username).to.equal('bar');
+      });
+    });
+
     it('allows sql logging of delete statements', function() {
       const UserDelete = this.sequelize.define('UserDelete', {
         name: Support.Sequelize.STRING,
         bio: Support.Sequelize.TEXT
       });
 
+      const logging = sinon.spy();
+
       return UserDelete.sync({ force: true }).then(() => {
         return UserDelete.create({ name: 'hallo', bio: 'welt' }).then(u => {
           return UserDelete.findAll().then(users => {
             expect(users.length).to.equal(1);
-            return u.destroy({
-              logging(sql) {
```

Comment:
This didn't verify that `logging` was actually called.

Reply:
Great catch! This reminded me of something I saw recently. I like how you fixed this, using sinon. Would you be willing to do something similar to refactor several other tests with a similar problem?

If you search for `expect(test).to.be.true` you will find 9 occurrences in tests, they all could be refactored to use sinon, would you like to do that?

(And perhaps, if the code gets too big, you could even create a support method to be reused around tests)

Reply:
(of course, this could be in a separate PR, if you're willing to do it - has nothing to do with this PR)